src/Modals.ts

```diff
@@ -200,9 +200,38 @@ export class Modals {
 
         // #settingsModal - showSettingsModal()
         $('#settingsModal').on('shown.bs.modal', function(){
+            $('#settingsModal').data('completed', false);
+            eagle.copyCurrentSettings()
             $('#settingsModalAffirmativeButton').focus();
         });
 
+        $("#settingsModalNegativeButton").on('click', function(){
+            eagle.cancelSettingChanges()
```

Comment:
I think this case will be automatically handled by the code on line 217.

If the user clicks the negative button, the 'hidden.bs.modal' event will occur, and 'completed' will be false. So eagle.cancelSettingChanges() will get called anyway.